extension/js/common/client-configuration.ts

```diff
@@ -83,6 +84,19 @@ export class ClientConfiguration {
     return this.clientConfigurationJson.enforce_keygen_expire_months;
   };
 
+  /**
+   * pass phrase session length to be configurable with client configuraiton
+   * default 4 hours
+   */
+  public getInMemoryPassPhraseSessionExpiration = (): number => {
+    let expireIn = 4 * 60 * 60;
+    if (this.clientConfigurationJson.in_memory_pass_phrase_session_length) {
+      // in_memory_pass_phrase_session_length min: 1, max: Int max value
+      expireIn = Math.max(1, Math.min(this.clientConfigurationJson.in_memory_pass_phrase_session_length, Number.MAX_VALUE));
+    }
+    return Date.now() + expireIn * 1000;
+  };
```

Comment:
My feedback here is just a hunch. I feel a little uneasy about this method returning the actual resulting date. I don't know exactly what makes me worry about it, since obviously we are not storing this information anywhere, so time passed between when it was queried, and when it was used, is no problem.

I guess it's a real nit-pick, but would it be a big change / would the code be messier if this method returned just amount of seconds, and the `Date.now() + expirationSeconds` would happen elsewhere?

Reply:
Looks like we (I) implemented `ExpirationCache.set` in a way that it accepts the date, not an offset, that's the reason

Reply:
I'd be quite content if this method was called as `Date.now() + clientConfiguration.getInMemoryPassPhraseSessionExpiration()` wherever it's used. Then it will be a bit more obvious.

Reply:
Hmm.. then we need to also accept email parameter in `ExpirationCache` constructor (or in `ExpirationCache.set ` method).
That seems a bit annoying to me.

Reply:
I meant just like this

![image](https://user-images.githubusercontent.com/6306961/190992399-5927acb9-e000-4e03-869f-3bebaab216d2.png)

And then also rename `getInMemoryPassPhraseSessionExpiration` to `getInMemoryPassPhraseSessionExpirationMs` for clarity.

Reply:
Aha, got it
Just changed